File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        
        for n in nums:
            if n not in count:
                count[n] = 0
            count[n] += 1
        
        
        freq = [[] for _ in range(len(nums) + 1)]
        
        for n, f in count.items():
            freq[f].append(n)
        
        
        res = []
        
        for arr in range(len(freq) -1, -1, -1):
            for n in freq[arr]:
                res.append(n)
                if len(res) == k:
                    return res
        
        
        
        

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    let map = {};
    let maxFreq = 0
    for (let num of nums) {
        if (!(num in map)) {
            map[num] = 0;
        }
        map[num]++;
        
        if (map[num] > maxFreq) {
            maxFreq = map[num];
        }
    }

    
    const arrFreq = Array(maxFreq + 1).fill().map(() => []);
    for (let key in map) {
        let val = map[key];
        arrFreq[val].push(key);
    }
        
    let res = [];
    
    for (let i = arrFreq.length - 1; i >= 0; i--) {
        arrFreq[i].forEach((num) => {
            res.push(num);
        });
        if (res.length === k) {
            return res
        }
    }
    
};